core/block_svg.ts

```diff
@@ -328,7 +328,14 @@ export class BlockSvg
     } else if (oldParent) {
       // If we are losing a parent, we want to move our DOM element to the
       // root of the workspace.
-      this.workspace.getCanvas().appendChild(svgRoot);
+      const draggingBlock = this.workspace
+        .getCanvas()
+        .querySelector('.blocklyDragging');
+      if (draggingBlock) {
+        this.workspace.getCanvas().insertBefore(svgRoot, draggingBlock);
```

Comment:
@gonfunko do you remember what case this was handling?

Reply:
Not offhand :/ Judging by the comment maybe it's when a block is disconnected from a stack?